jax/experimental/jax2tf/jax2tf.py

```diff
@@ -1330,31 +1330,93 @@ def reducer(arg1: TfVal, arg2: TfVal) -> TfVal:
-  function defining the initial value of the reduction depending on the dtype of the
-  operand.
+# _try_tf_max_pool returns a Tensor when it succeeds, or a string describing why
+# it did not succeed otherwise.
+def _try_tf_max_pool(operand, window_dimensions, window_strides, padding,
+                     base_dilation, window_dilation) -> Union[str, TfVal]:
+  # TODO(bchetioui): this function is not exhaustive wrt which
+  # reduce_window_max cases can be translated into a call to max_pool. Further
+  # investigation is needed to fully flesh it out.
+
+  # Contrarily to the main path, tf.int8 is actually a valid type for
+  # tf.nn.max_pool.
+  if operand.dtype in [tf.bool, tf.uint32, tf.uint64, tf.complex64,
+                       tf.complex128]:
+    return f"tf.nn.max_pool does not support operands of type {operand.dtype}"
+  has_batch_dim = window_dimensions[0] == 1
+  has_channel_dim = window_dimensions[-1] == 1
+  nb_spatial_dimensions = len(operand.shape) - has_batch_dim - has_channel_dim
+  if nb_spatial_dimensions < 1 or nb_spatial_dimensions > 3:
+    return ("TensorFlow can only handle max pooling for arrays with 1, 2, or "
+            "3 spatial dimensions")
+  # TODO(bchetioui): does a simple conversion with another base dilation exist?
+  if list(base_dilation) != [1] * len(operand.shape):
+    return "Unimplemented support for base dilation"
+  # TODO(bchetioui): does a simple conversion with another window_dilation
+  # exist? The whole story seems similar to convolution.
+  if list(window_dilation) != [1] * len(operand.shape):
+    return "Unimplemented support for window dilation"
+  if list(padding) != [(0, 0)] * len(operand.shape):
+    return "Unimplemented support for padding"
+  # ReduceWindow in XLA takes an array of rank N as a parameter, but
+  # tf.nn.max_pool takes an array of rank N+2, with a default shape of the
+  # form [batch_size] + input_spatial_shape + [num_channels]
+  tf_operand = operand
+  tf_window_dimensions = list(window_dimensions)
+  tf_window_strides = list(window_strides)
+  if not has_batch_dim:
+    tf_operand = tf.expand_dims(tf_operand, 0)
+    tf_window_dimensions = [1] + tf_window_dimensions
+    tf_window_strides = [1] + tf_window_strides
+  if not has_channel_dim:
+    tf_operand = tf.expand_dims(tf_operand, -1)
+    tf_window_dimensions.append(1)
+    tf_window_strides.append(1)
+  tf_data_format = 'N' + 'DHW'[-nb_spatial_dimensions:] + 'C'
+  tf_padding = 'VALID'
+  result = tf.nn.max_pool(tf_operand, tf_window_dimensions, tf_window_strides,
+                          tf_padding, tf_data_format)
+  if not has_batch_dim:
+    result = tf.squeeze(result, 0)
+  if not has_channel_dim:
+    result = tf.squeeze(result, -1)
+  return result
+
+def _specialized_reduce_window(reducer, identity, operand, window_dimensions,
+                               window_strides, padding, base_dilation,
+                               window_dilation, _in_avals, _out_aval, *,
```

Comment:
I am very surprised by the position of the '*'. I expected that it should be right after `operand`, because all the others are parameters to the primitive.

Reply:
They are, and it would probably make more sense to do it like that, you're right. I'll make that change shortly.